src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -138,5 +141,52 @@ def calculate_prices(key, functor):
                 self._update_tree_summary(child)
 
     def _update_bills(self, area, past_market_types):
-        result = energy_bills(area, past_market_types)
-        return OrderedDict(sorted(result.items()))
+        result, external_trades = energy_bills(area, past_market_types, {})
+        return OrderedDict(sorted(result.items())), external_trades
+
+    def _calculate_redis_bills(self, area, energy_bills, external_trades):
+        from copy import deepcopy
```

Comment:
Any reason for this lazy import?

Reply:
You are right, this has to be global, moving this